k8s/cmd/commands/util/namespace.go

```diff
@@ -0,0 +1,39 @@
+/*
+* Copyright 2020-2023 Alibaba Group Holding Limited.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package util
+
+import (
+	"github.com/v6d-io/v6d/k8s/cmd/commands/flags"
+	"github.com/v6d-io/v6d/k8s/pkg/log"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CreateNamespaceIfNotExist creates namespace if it does not exist
+func CreateNamespaceIfNotExist(c client.Client) {
+	namespace := corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: flags.Namespace,
+		},
+	}
+
+	if flags.CreateNamespace {
+		if err := CreateIfNotExists(c, &namespace); err != nil {
+			log.Fatal(err, "Expects no positional arguments")
+		}
+	}
+}
```

Comment:
It should be ok to put it into `util/client.go`, as we already have many utils e.g., create/delete, it in.

Reply:
Do you mean while creating a resource if the namespace doesn't exist, then create a namespace first?

Reply:
I mean no need to put the "CreateNamespace" function into a seperate file, as we already have some similar things in client.go

https://github.com/v6d-io/v6d/blob/main/k8s/cmd/commands/util/client.go

Reply:
Got it and resolved.  Also, the reference doc updated .